app/models/subscriber.rb

```diff
@@ -0,0 +1,32 @@
+class Subscriber
+  def self.for_discussion(discussion)
+    new(discussion)
+  end
+
+  def initialize(discussion)
+    @discussion = discussion
+    @subscriptions = Subscription.where(discussion_id: discussion.id)
+  end
+
+  def subscribe(users)
+    users.each do |user|
+      create_subscription_for_user(user)
+    end
+  end
+
+  private
+
+  attr_reader :discussion, :subscriptions
+
+  def create_subscription_for_user(user)
+    return if user_already_subscribed?(user)
+
+    Subscription.create(user: user, discussion: discussion)
+  end
+
+  def user_already_subscribed?(user)
+    subscriptions.any? do |subscription|
+      subscription.user_id == user.id
+    end
+  end
```

Comment:
My idea was to make this check without calling to the db, so the uniqueness validation would be a no go. You would have to call the db, just to get an error back.

As for the `where` I'm on the fence here. The main goal of this pr was to remove an N+1 query problem, and this `where` would be called for each user. I was trying to avoid calls made for each user, unless strictly necessary, as the creation of the subscription.

What do you think?